Gems/Archive/Code/Include/Archive/ArchiveInterfaceStructs.h

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/base.h>
+
+#include <AzCore/IO/Path/Path_fwd.h>
+#include <AzCore/std/containers/array.h>
+#include <AzCore/std/containers/span.h>
+#include <AzCore/std/limits.h>
+
+#include <Compression/CompressionInterfaceStructs.h>
+
+namespace Archive
+{
+    inline namespace literals
+    {
+        constexpr AZ::u64 operator"" _kib(AZ::u64 value);
+        constexpr AZ::u64 operator"" _mib(AZ::u64 value);
+        constexpr AZ::u64 operator"" _gib(AZ::u64 value);
+    }
+
+    //! Represents the default block size for the Archive format
+    //! It will be 2 MiB until their is data that proves
+    //! that a different block size is ideal
+    constexpr AZ::u64 ArchiveDefaultBlockSize = 2  * (1 << 20);
+    //! The alignment of blocks within an archive file
+    //! It defaults to 512 bytes
+    constexpr AZ::u64 ArchiveDefaultBlockAlignment = 512;
+
+    //! Sentinel which indicates the last entry in the deleted File Path Index list
+    constexpr AZ::u32 DeletedPathIndexSentinel = AZStd::numeric_limits<AZ::u32>::max();
+
+    //! Sentinel which indicates the value written to the last block to indicate
+    //! there are no further deleted blocks afterwards
+    constexpr AZ::u64 DeletedBlockOffsetSentinel = AZStd::numeric_limits<AZ::u64>::max();
+
+    //! Fixed size Header struct for the Archive format
+    //! This suitable for directly reading the archive header into
+    struct ArchiveHeaderSection
+    {
+        ArchiveHeaderSection();
+
+        //! O3DE only runs on little endian machines
+        //! Therefore the bytes are added in little endian order
+        //! The Magic Identifier for the archive format is "O3AR" for O3DE Archive
+        //! offset = 0
+        const AZ::u32 m_magicBytes{ 'O' | ('3' << 8) | ('A' << 16) | ('R' << 24) };
+
+        //! Version number of archive format. Supports up to 2^16 revisions per entry
+        //! offset = 4
+        AZ::u16 m_minorVersion{};
+        AZ::u16 m_majorVersion{};
+        AZ::u16 m_revision{};
+
+        //! reserved for future memory configurations
+        //! default layout is 2Mib blocks with 512 byte borders
+        //! offset = 10
+        AZ::u16 m_layout{};
+
+        //! Represents the The number of files stored within the Archive
+        //! Caps out at (2^25) or ~33 million files that can be represented
+        //! offset = 12
+        AZ::u32 m_fileCount{};
```

Comment:
What do you think about including more details about this limit?
```suggestion
        //! Represents the The number of files stored within the Archive
        //! Caps out at (2^25) or ~33 million files that can be represented
        //! Up to 2^32 can be stored, but is limited to 2^25 because around 640MiB of uncompressed data will need to be loaded into memory for an archive of containing ~33million files, and memory requirements would increase if the not limited.
        //! offset = 12
        AZ::u32 m_fileCount{};
```